aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py

```diff
@@ -361,6 +362,9 @@ def _set_remote_type_and_identifier(span: ReadableSpan, attributes: BoundedAttri
         elif is_key_present(span, AWS_STREAM_NAME):
             remote_resource_type = _NORMALIZED_KINESIS_SERVICE_NAME + "::Stream"
             remote_resource_identifier = _escape_delimiters(span.attributes.get(AWS_STREAM_NAME))
+        elif is_key_present(span, AWS_CONSUMER_ARN):
+            remote_resource_type = _NORMALIZED_KINESIS_SERVICE_NAME + "::StreamConsumer"
+            remote_resource_identifier = _escape_delimiters(span.attributes.get(AWS_CONSUMER_ARN))
```

Comment:
Is it possible that both stream name and consumer arn are present? Which is the right one to use in that case, why?

Reply:
I checked all APIs that uses consumerARN, there is no case that both stream name and consumer arn are present. So we should be safe in this case. But it is a good callout. We need to double check to other upcoming services.